### PR TITLE
Expiration date update

### PR DESCRIPTION
**Pull request by MVarshini**
PBECNH-1311

Expiration date update

### CODE REVIEW

**Review by dbutenhof — COMMENTED** [review]
I deployed from this branch, and it doesn't seem to fix the problem for me. I still lose focus on the first click...

[Screencast from 2024-01-10 09-14-01.webm](https://github.com/distributed-system-analysis/pbench/assets/61761529/0586baa5-d9ad-449e-b949-256f9b377356)

File: dashboard/src/modules/components/OverviewComponent/SavedRunsComponent.jsx [hunk 1]
```diff
@@ -204,7 +204,7 @@ const SavedRunsComponent = () => {
                           updateTblValue(val, NAME_KEY, item.resource_id)
                         }
                         toggleEdit={toggleEdit}
-                        onDateSelect={(_event, str) =>
+                        onDateSelect={(str) =>
                           updateTblValue(
```
Comment (webbnh) [inline dashboard/src/modules/components/OverviewComponent/SavedRunsComponent.jsx:208]: The call to `updateTblValue()` is going to result in a call to `dispatch()`...will that result in the page being re-rendered, which will result in the input focus being reset?  (If so, that would explain what we're seeing...although, it doesn't suggest a _solution_ for the problem....)

**Comment by dbutenhof** [comment]
Since you'd done a rebase on this branch a few days ago, I'd been meaning to get around to pulling the branch to check whether it was working now; and I'm still seeing the same behavior. The focus doesn't stick, and the calendar picker doesn't actually change the date.

Could this be related to Webb's comment about the dispatch behavior? Or is something else going wrong? And, is this only me (and perhaps Chrome)? Is it working for you?

(FYI, I did a full `runlocal` and connected to `https://localhost:8443/dashboard` ... I suppose that if you're using dev mode, there might be a difference?)

**Comment by MVarshini** [comment]
As there were problems with `PatternFly Datepicker` in production replaced it with `react-datepicker`.  This would resolve the problem that we encounter in editing the expiration date.

File: dashboard/src/modules/components/OverviewComponent/SavedRunsComponent.jsx (hunk 1 above)
Comment (MVarshini) [reply dashboard/src/modules/components/OverviewComponent/SavedRunsComponent.jsx:208]: The input focus reset is the problem with editing the dataset name. I'll try to fix it in the different PR.

**Review by dbutenhof — COMMENTED** [review]
The "from" calendar popup and text entry are working, so that's great. However the "to" calendar and text entry remain grayed out and inactive, along with the Update button.
image

**Review by webbnh — APPROVED** [review]
Once Dave's concern has been addressed, I'm good with this merging, with the expectation that the fix to [the other problem](https://github.com/distributed-system-analysis/pbench/pull/3597#discussion_r1447765624) will come in its own PR.

**Review by dbutenhof — APPROVED** [review]
Varshini wants to fix the to/update problem separately, so let's go.

**Pull request merged**